<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>README</title>
</head>
<body>

<h1> Table of contents</h1>
Brief Intro
BEsingle.py
BEmain.py

BE-FF is a web-based tool that receives SNV data and matches suitable BEs to correct the
variation. The code for the online tool is available here, as well as batch-mode code that may be used
to generate results for a large number of SNV's.
The main 3 scripts are:
1. BEsingle.py:
This is the main script of the website. It receives a single SNV, via 3 possible methods:
a. Manually entered by user: Here you must enter a 51-nt long DNA sequence, 25 nt upstream to the mutation and
25 nt downstream, as well as the variation and the reading frame.
b. Fetched from given rsID: Enters a known rsID. You will then be presented with a table containing all the
possible variations. Once you selects one of the options, it will automatically be inserted into the format in (a).
c. Fetched by genomic coordinates : You may select a genome, chromosome number, mutation position, variation, and reading frame,
which will be inserted into the format in (a).
**Note: Only one of these methods is required each time. **

You may also use the 'Advanced Options' button and create you own BE. This base editor will appear as
'User customized BE' in the final results

Press the submit button to show the results. You will be presented with two tables:
1. Table of  base editors that will correct this variance and no other bases around it.
2. Table of base editors that will correctly edit the SNP. However it may also edit other flanking nucleotides.
Such changes are synonymous substitutions and the resulted amino acid sequence will probably be as good as the reference sequence.





and returns all the possible
In the scripts folder, open the file 'BEMain.py'. In lines 506 and 522,
change the name of the file to your file's name. The current file, 'snps_with_clinvar2' is available
in the Scripts folder as well. Run the script. Note: You may need to download BioSeq if not already
installed on your computers. The output is a CSV file named 'full results.csv'
AUTHOR: Shiri Almog , shirialmog1@gmail.com
</body>
</html>